src/translator/quality_estimator.cpp

```diff
@@ -0,0 +1,294 @@
+
+  const size_t lrSize =
+      (scale_.means.size() + scale_.stds.size() + coefficients_.size()) * sizeof(float) + sizeof(intercept_);
+
+  Header header = {BINARY_QE_MODEL_MAGIC, lrParametersDims};
+  marian::bergamot::AlignedMemory memory(sizeof(header) + lrSize);
+
+  char* buffer = memory.begin();
+
+  memcpy(buffer, &header, sizeof(header));
+  buffer += sizeof(header);
+
+  for (const float std : scale_.stds) {
+    memcpy(buffer, &std, sizeof(std));
+    buffer += sizeof(std);
+  }
+
+  for (const float mean : scale_.means) {
+    memcpy(buffer, &mean, sizeof(mean));
+    buffer += sizeof(mean);
+  }
+
+  for (size_t i = 0; i < lrParametersDims; ++i) {
+    const float coefficient = coefficients_[i];
+    memcpy(buffer, &coefficient, sizeof(coefficient));
+    buffer += sizeof(coefficient);
+  }
+
+  memcpy(buffer, &intercept_, sizeof(intercept_));
+  buffer += sizeof(intercept_);
+
+  return memory;
+}
+
+void LogisticRegressorQualityEstimator::computeQualityScores(const Histories& histories, Response& response) const {
+  size_t sentenceIndex = 0;
+
+  for (const auto& history : histories) {
+    const std::vector<float> logProbs = std::get<1>(history->top())->tracebackWordScores();
+    response.qualityScores.push_back(computeSentenceScores(logProbs, response.target, sentenceIndex));
+
+    ++sentenceIndex;
+  }
+}
+
+Response::SentenceQualityEstimate LogisticRegressorQualityEstimator::computeSentenceScores(
+    const std::vector<float>& logProbs, const AnnotatedText& target, const size_t sentenceIdx) const
+
+{
+  const std::vector<SubwordRange> wordIndexes = mapWords(logProbs, target, sentenceIdx);
+
+  const std::vector<float> wordScores = predict(extractFeatures(wordIndexes, logProbs));
+
+  const float sentenceScore =
+      std::accumulate(std::begin(wordScores), std::end(wordScores), float(0.0)) / wordScores.size();
+
+  return {wordScores, subwordToWords(wordIndexes, target, sentenceIdx), sentenceScore};
+}
+
+std::vector<float> LogisticRegressorQualityEstimator::predict(const Matrix& features) const {
+  std::vector<float> scores(features.rows);
+
+  for (int i = 0; i < features.rows; ++i) {
+    for (int j = 0; j < features.cols; ++j) {
+      scores[i] += features.at(i, j) * coefficientsByStds_[j];
+    }
+  }
+
+  /// Applies the linear model followed by a sigmoid function to each element
+
+  for (int i = 0; i < features.rows; ++i) {
+    scores[i] = std::log(1 - (1 / (1 + std::exp(-(scores[i] - constantFactor_ + intercept_)))));
+  }
+
+  return scores;
+}
+
+LogisticRegressorQualityEstimator::Matrix LogisticRegressorQualityEstimator::extractFeatures(
+    const std::vector<SubwordRange>& wordIndexes, const std::vector<float>& logProbs) const {
+  if (wordIndexes.empty()) {
+    return std::move(Matrix(0, 0));
+  }
+
+  Matrix features(wordIndexes.size(), /*numFeatures =*/4);
+  size_t featureRow = 0;
+  const size_t I_MEAN{0}, I_MIN{1}, I_NUM_SUBWORDS{2}, I_OVERALL_MEAN{3};
```

Comment:
Same comment as above for `numFeatures`, `I_MEAN`, `I_MIN`, `I_NUM_SUBWORDS`, `I_OVERALL_MEAN`

Reply:
Just added a brief documentation here 
c762e3b

See if you agree with it :)